cdap-notifications/src/main/java/co/cask/cdap/notifications/service/BasicNotificationContext.java

```diff
@@ -51,7 +51,7 @@ public boolean execute(TxRunnable runnable, TxRetryPolicy policy) {
         final TransactionContext context = new TransactionContext(transactionSystemClient);
         try {
           context.start();
-          runnable.run(new DynamicDatasetContext(namespaceId, context, dsFramework,
+          runnable.run(new DynamicDatasetContext(namespaceId, context, null, dsFramework,
```

Comment:
Because there is no metrics context here. At least I am not sure what the context should be.